test/instrumentation/modules/http/github-423.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+require('../../../..').start({
+  serviceName: 'test',
+  captureExceptions: false
+})
+
+var http = require('http')
+var zlib = require('zlib')
+var fs = require('fs')
+var got = require('got')
+var test = require('tape')
+
+test('https://github.com/elastic/apm-agent-nodejs/issues/423', function (t) {
+  // Start dummy remote server to fetch gzip'ed data from
+  var remote = http.createServer(function (req, res) {
+    res.setHeader('Content-Encoding', 'gzip')
+    fs.createReadStream(__filename).pipe(zlib.createGzip()).pipe(res)
+  })
+
+  remote.listen(function () {
+    var port = remote.address().port
+    var url = 'http://localhost:' + port
+
+    // Start simple server that performs got-request on every request
+    var server = http.createServer(function (req, res) {
+      got(url).then(function (response) {
+        t.equal(response.body[0], '\'', 'body should be uncompressed')
```

Comment:
I'd verify more than just the first character.

Reply:
If it's gzipped, it will always start with a few magic gzip bytes, so it can never be `'`. But I could just read the entire file into memory up top and do a complete comparison - that way it's good for sure 😃

Reply:
Hmm actually the downside of that is that the test output will display the binary gzipped data if the test fails when mimic-response isn't correctly patched. And that might mess with your shell. The first magic byte in the gzip header is `\x1f` and will always be displayed as such, so there's no danger of messing up the terminal in that case.

Here's how it looks if I display it all:

![image](https://user-images.githubusercontent.com/10602/42239121-3ca4a18e-7f03-11e8-833b-5b7602aa5736.png)

Reply:
Continuing the conversation with my self: Actually the dangerous bytes are probably all escaped - just as `\x1f`, so I guess my concern is invalid 😅

Reply:
Ok, ended up testing just the first line and added a test for the total length as well. That should hopefully cover everything 😃